src/app/my-journeys-listing/my-journey/my-journey.page.ts

```diff
@@ -0,0 +1,116 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { LoaderService } from 'src/app/services/loader/loader.service';
+import { ProfileService } from 'src/app/services/profile/profile.service';
+import { ProjectsApiService } from 'src/app/services/projects-api/projects-api.service';
+import urlConfig from 'src/app/config/url.config.json';
+import { ToastService } from 'src/app/services/toast/toast.service';
+
+@Component({
+  selector: 'app-my-journey',
+  templateUrl: './my-journey.page.html',
+  styleUrls: ['./my-journey.page.scss'],
+})
+export class MyJourneyPage  {
+  headerConfig={
+    title:"MY_JOURNEY",
+    showBackButton: true,
+  }
+  programId:any;
+  programName:any;
+  constructor(
+    private router:Router,
+    private route:ActivatedRoute,
+    private profileService: ProfileService,
+    private loaderService: LoaderService,
+    private projectsApiService: ProjectsApiService,
+    private toastService: ToastService
+  ) {
+    const navigation = this.router.getCurrentNavigation();
+    this.programName = navigation?.extras.state?.['data'];
+    this.route.params.subscribe(param=>{
+      this.programId = param['id'];
+  })
+  }
+
+  myJourneyInprogress :any[]=[];
+  myJourneyCompleted: any[]=[];
+  page = 1;
+  limit = 15;
+  disableLoading: boolean = false;
+  pageConfig: any = {};
+  profilePayload: any = {};
+  selectedSegment: string = 'inProgress';
+
+  segmentChanged(event: any) {
+    this.selectedSegment = event.detail.value;
+    this.myJourneyInprogress=[];
+    this.myJourneyCompleted=[];
+    this.page = 1;
+    this.limit = 15;
+    this.getProjects(this.selectedSegment)
```

Comment:
Why are we passing the selectedSegment even though its globally stored
This value can be directly accessed from that function right?

Reply:
done

Reply:
I dont see any changes made here?

Reply:
the content changes upon the segment is changed.

Reply:
But do we need to pass it while calling this function this.getProjects(this.selectedSegment)

Reply:
yes